src/classes/BDI_ContactService.cls

```diff
@@ -309,13 +309,17 @@ public with sharing class BDI_ContactService {
     */
     private void associateDIsWithMatchedOrNewContacts(String strCx) {
         Integer iCx = strCx == 'Contact1' ? 1 : 2;
+        Boolean modelIsOneToOne = CAO_Constants.isOneToOne();
 
         for (DataImport__c di : bdi.listDI) {
             // skip di's that already have an error
             if (di.Status__c == BDI_DataImport_API.bdiFailed)
                 continue;
             List<String> listDiKey = listDIKeyCx(di, strCx);
             Contact con = ContactFromDiKeys(listDiKey);
+            Boolean bothContactsSpecified = (strCx == 'Contact1' && isContactSpecified(di, 'Contact2')) || 
```

Comment:
Is there a scenario where you'd want to let Contact2 get through if Contact1 is blank?

Reply:
If so, it would be good to add a test to confirm that it will work when Contact2 alone is populated